slack/webhook/client.py

```diff
@@ -0,0 +1,117 @@
+import json
+import logging
+from http.client import HTTPResponse
+from typing import Dict, Union
+from urllib.error import HTTPError
+from urllib.request import Request, urlopen
+
+from slack.errors import SlackRequestError
+from .webhook_response import WebhookResponse
+from ..web import convert_bool_to_0_or_1, get_user_agent
+from ..web.classes.blocks import Block
+
+
+class WebhookClient:
+    logger = logging.getLogger(__name__)
+
+    def __init__(
+        self, url: str, default_headers: Dict[str, str] = {},
+    ):
+        """API client for Incoming Webhooks and response_url
+        :param url: a complete URL to send data (e.g., https://hooks.slack.com/XXX)
+        :param default_headers: request headers to add to all requests
+        """
+        self.url = url
+        self.default_headers = default_headers
+
+    def send(
+        self, body: Dict[str, any], additional_headers: Dict[str, str] = {},
+    ) -> WebhookResponse:
+        """Performs a Slack API request and returns the result.
+        :param body: json data structure (it's still a dict at this point),
+            if you give this argument, body_params and files will be skipped
+        :param additional_headers: request headers to append only for this request
+        :return: API response
+        """
+        body = convert_bool_to_0_or_1(body)
+        self._parse_blocks(body)
+        if self.logger.level <= logging.DEBUG:
```

Comment:
TODO: do this debug logging after generating a JSON payload  in `_perform_http_request`